### PR TITLE
i18n: add ability to omit POST-Creation-Date from being generated

### LINKED ISSUE #5887: I18N extraction: make POT-Creation-Date optional
The `POT-Creation-Date` changes everytime the command `i18n extract` is run.

This creates headaches when working with multiple feature branches, each getting their own translations and constantly changes the same line on all files.

It's in general also annoying to have this noise in the diff.

Suggestion: make the addition of this timestamp optional or, if required for other tools, allow to fixate it at e.g. 1970-01-01

### PR DESCRIPTION
**Pull request by mfn**
Fixes #5887

### CODE REVIEW

File: lib/Cake/Console/Command/Task/ExtractTask.php [hunk 1]
```diff
@@ -346,6 +346,10 @@ public function getOptionParser() {
 		))->addOption('extract-core', array(
 			'help' => __d('cake_console', 'Extract messages from the CakePHP core libs.'),
 			'choices' => array('yes', 'no')
+		))->addOption('omit-pot-creation-date', array(
```
Comment (dereuromark) [inline lib/Cake/Console/Command/Task/ExtractTask.php]: `omit-creation-date` would be a little bit shorter maybe.

Comment (ADmad) [inline lib/Cake/Console/Command/Task/ExtractTask.php]: Something shorted like `skip-date` would be better

Comment (mfn) [inline lib/Cake/Console/Command/Task/ExtractTask.php]: Great, changed! I'm notoriously bad at naming things ...

**Comment by mfn** [comment]
I've fixed the tests now too (I don't see how the cachetest relates to my changes; seems like a timing race condition).

Before considering for merge I'd probably clean this up and squash all commits into one.

Q about the tests: wouldn't it be better to set all the default values normally set by `getOptionParser()` into `setUp()` instead of trying to set up the class for each test requiring `->execute()` ?

**Comment by markstory** [comment]
Why don't we just remove the date entirely? It doesn't seem like it is that useful

**Comment by mfn** [comment]
Well, :+1:  from me. I also don't see its value. I take it given that every project today uses a VCS anyway, but that may just be me.

File: lib/Cake/Console/Command/Task/ExtractTask.php (hunk 1 above)
Comment (johannesnagl) [inline lib/Cake/Console/Command/Task/ExtractTask.php]: i like the `skip-date` option naming :)

**Comment by AD7six** [comment]
I would also prefer to just get rid of the date, and remove the source of noise.

**Comment by markstory** [comment]
Yeah I am for removing the date entirely as well. Do you want to update your pull request @mfn? If not I can remove the troublesome code.

**Comment by mfn** [comment]
Updated/rebased branch!

**Comment by markstory** [comment]
Thanks :+1:

**Pull request merged**